clang/lib/CIR/CodeGen/CIRGenDecl.cpp

```diff
@@ -76,7 +86,8 @@ CIRGenFunction::buildAutoVarAlloca(const VarDecl &D) {
       // be done as part of lowering down to LLVM.
       if ((!getContext().getLangOpts().OpenCL ||
            Ty.getAddressSpace() == LangAS::opencl_constant) &&
-          (!NRVO && !D.isEscapingByref() && CGM.isTypeConstant(Ty, true)))
+          (!NRVO && !D.isEscapingByref() &&
+           CGM.isTypeConstant(Ty, true, false)))
```

Comment:
add `/*name_of_parm=*/` to both `true` and `false` to make it more explicit what this is about. Please apply this change to all other `isTypeConstant()` changed in this PR.